sdk/textanalytics/ai-text-analytics/samples/javascript/alternativeDocumentInput.js

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Demonstrates how to use object inputs as alternatives to
+ * string arrays in client methods
+ */
+
+const { TextAnalyticsClient, AzureKeyCredential } = require("@azure/ai-text-analytics");
+
+// Load the .env file if it exists
+const dotenv = require("dotenv");
+dotenv.config();
+
+// You will need to set this environment variables or edit the following values
+const endpoint = process.env["ENDPOINT"] || "<cognitive services endpoint>";
+const apiKey = process.env["TEXT_ANALYTICS_API_KEY"] || "<api key>";
+
+/**
+ * DetectLanguageInput objects allow for specification of country hints on a
+ * document-by-document basis.
+ *
+ * When using DetectLanguageInput, the `id` field is required and must be unique
+ * for each document in a given request.
+ */
+const detectLanguageInputs = [
+  { id: "0", countryHint: "us", text: "I had the best day of my life." },
+  { id: "1", countryHint: "us", text: "This was a waste of my time. The speaker put me to sleep." },
+  { id: "2", countryHint: "us", text: "No tengo dinero ni nada que dar..." },
+  {
+    id: "3",
+    countryHint: "fr",
+    text: "L'hôtel n'était pas très confortable. L'éclairage était trop sombre."
+  }
+];
+
+/**
+ * TextDocumentInput objects are used by all methods except for `detectLanguage`.
+ * The allow for specification of input language on a document-by-document basis.
+ *
+ * Like `DetectLanguageInput`, the `id` field is required and must be unique for
+ * each document in the request.
+ */
+const textDocumentInputs = [
```

Comment:
It doesn't seem necessary to include comments about TextDocumentInput not working for detect_language etc since you're just using a dictionary. Also suggest changing the name of the variable textDocumentInptuts since you're not using TextDocumentInput here, maybe `extractKeyPhrasesInputs`

Reply:
Hmm. This is a side effect of the fact that in TS these are strongly-typed. So in TS the signature is `const textDocumentInputs : TextDocumentInput[] = [ ... ];`, so I felt it was appropriate to elaborate, there.

In JS the types are erased, so I will tweak the text so that it makes more sense in this context.